projects/fal/src/fal/api.py

```diff
@@ -754,10 +758,20 @@ def __call__(self, *args, **kwargs) -> None:
                 f"[warning] {self._func.__name__} function is served with no arguments."
             )
 
-        from uvicorn import run
+        from fastapi import FastAPI
+        from starlette_exporter import PrometheusMiddleware, handle_metrics
+        from uvicorn import Config, Server
 
         app = self.build_app()
-        run(app, host="0.0.0.0", port=8080)
+        app.add_middleware(PrometheusMiddleware, prefix="http")
+        metrics_app = FastAPI()
+        metrics_app.add_route("/metrics", handle_metrics)
+        asyncio.get_event_loop().run_until_complete(
```

Comment:
i think you might need a new event loop here, and then run it (there might not be one set).

Reply:
`get_event_loop` will start a new loop if one is not already running